### PR TITLE
Use label for trajectory

### LINKED ISSUE #342: Refactor: remove `metric_type` and introduce `label`

### LINKED ISSUE #344: Fix line order for green technologies
After the re-ordering of lines for trajectory the lines are assigned in wrong order causing the main line to be a dashed line.

image

### PR DESCRIPTION
**Pull request by MonikaFu**
Closes #342 
Closes #344 

In this PR I remove the need for column 'metric0' and make the plotting label use column 'label' for annotations and 'metric' in combination with `is_scenario()` for everything else. 

The plot does not have 'pretty labels' anymore. That feature will be introduced back in a quick plot.

### CODE REVIEW

File: R/plot_trajectory.R [hunk 1]
```diff
@@ -231,35 +231,31 @@ get_ordered_scenario_colours <- function(n) {
...
-        is_scenario(.data$metric0) ~ toupper(as.character(.data$metric)),
-        TRUE ~ to_title(as.character(.data$metric))
-      )
-    )
+    mutate(value = .data$production)
+
+  if (!("label" %in% names(out))) {
+    out <- out %>%
+      mutate(label = .data$metric)
+  }
```
Comment (MonikaFu) [inline R/plot_trajectory.R:243]: The above could potentially be refactored to sth more succinct.

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # r2dii.plot (development version)
 
+* `plot_trajectory()` now plots legend labels from `metric` "as is" (#346).
```
Comment (maurolepore) [inline NEWS.md:3]: We need to remember to report user-facing changes.

File: R/plot_trajectory.R [hunk 3]
```diff
@@ -69,15 +69,15 @@ plot_trajectory_impl <- function(data) {
...
   p <- p + ggrepel::geom_text_repel(
     data = lines_end,
-    aes(label = .data$metric, segment.color = .data$metric),
+    aes(label = .data$label, segment.color = .data$metric),
     direction = "y",
     color = "black",
     size = 3.5,
     alpha = 1,
-    nudge_x = if_else(is_scenario(lines_end$metric0), 0.6, 0.1),
+    nudge_x = if_else(is_scenario(lines_end$metric), 0.6, 0.1),
```
Comment (maurolepore) [inline R/plot_trajectory.R:77]: Awesome! I created an issue to do the same elsewhere: #350

File: R/plot_trajectory.R [hunk 4]
```diff
@@ -103,12 +103,12 @@ value_span <- function(data) {
...
 line_colours <- function(data) {
   linecolours <- c("black", "black", "gray", "grey46", "black")
-  c(scenario_lines(data)$colour, linecolours[1:lines_n(data)])
+  c(scenario_lines(data)$colour, rev(linecolours[1:lines_n(data)]))
 }
 
 line_types <- function(data) {
   linetypes <- c("solid", "dashed", "solid", "solid", "twodash")
-  c(rep("solid", nrow(scenario_lines(data))), linetypes[1:lines_n(data)])
+  c(rep("solid", nrow(scenario_lines(data))), rev(linetypes[1:lines_n(data)]))
```
Comment (maurolepore) [inline R/plot_trajectory.R:111]: Thanks!

File: tests/testthat/test-plot_trajectory.R [hunk 5]
```diff
@@ -221,3 +212,23 @@ test_that("with no data to remove does not inform about removing rows", {
     expect_no_message() # No other message should bubble up
   options(restore)
 })
+
+test_that("'metric' in plot data is a factor with 'projected' as last element",{
```
Comment (maurolepore) [inline tests/testthat/test-plot_trajectory.R:216]: Awesome!

File: tests/testthat/test-plot_trajectory.R [hunk 6]
```diff
@@ -221,3 +212,23 @@ test_that("with no data to remove does not inform about removing rows", {
...
+  plot <- plot_trajectory(data)
+  expect_equal(levels(plot$data$metric)[nlevels(plot$data$metric)], "projected")
+
+  brown <- "oil"
+  data <- filter(market_share, technology == brown, region == first(region))
+  plot <- plot_trajectory(data)
+  expect_equal(levels(plot$data$metric)[nlevels(plot$data$metric)], "projected")
+})
+
+test_that("does not modify `metric`", {
```
Comment (maurolepore) [inline tests/testthat/test-plot_trajectory.R:227]: Added this to avoid regression.

**Review by maurolepore — APPROVED** [review]
LGTM!

**Pull request merged**